# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.h
# Commit: 7c94d37c446e
# Full Hash: 7c94d37c446ea0d996bf6e6e04eb66101e126b5c
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - WebXR compositing fixes. r=imanol
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78117
# ==============================================================================

diff -r 204b899f436d -r 7c94d37c446e dom/canvas/ClientWebGLContext.h
--- a/dom/canvas/ClientWebGLContext.h	Thu Jun 11 06:37:35 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.h	Thu Jun 11 06:37:36 2020 +0000
@@ -974,7 +974,6 @@
   layers::LayersBackend GetCompositorBackendType() const;
 
   bool mCapturedFrameInvalidated = false;
-  UniquePtr<layers::SurfaceDescriptor> mFrontBufferDesc;
 
   // -------------------------------------------------------------------------
   // WebGLRenderingContext Basic Properties and Methods
@@ -995,8 +994,9 @@
   }
   void GetContextAttributes(dom::Nullable<dom::WebGLContextAttributes>& retval);
 
-  void Present();
-  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(layers::TextureType);
+  void Present(WebGLFramebufferJS*, layers::TextureType);
+  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(WebGLFramebufferJS*,
+                                                  layers::TextureType);
   RefPtr<gfx::SourceSurface> GetFrontBufferSnapshot() override;
 
  private: